determined-ai/determined
# feat: detached mode v0.

## Description

minimal detached mode feature. for now, intended for internal use only and gathering feedback.

Currently, the APIs are placed under `determined.experimental.detached`, with detached context init mimicking the standard core API init. Current long-term plan is to merge this detached context init into the one and only core api init. This should ease the user migration between detached / attached runs. 

## Test Plan

- Try running the example scripts (`detached_2_hp_search.py` or `detached_3_torch_distributed.py`) outside of determined or in a notebook. 
- See the metrics in the UI. 
- Extend the examples and build your own detached experiment.

## Commentary (optional)

- you may need to configure `Determined()` client (master url and credentials) when running off-cluster.
- The examples print the output experiment URL. This URL may not match the master URL you need to paste into your browser if you're running the code on a remote node / notebook.
- No experiment state tracking.
- No determined hp search, bring your own hp search library.

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
